lexer/stateful/pattern_test.go

```diff
@@ -0,0 +1,51 @@
+package stateful
+
+import (
+	"log"
+	"regexp/syntax"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
```

Comment:
FYI, somewhat counter-intuitively, `assert.*` functions do not fail the test immediately, they continue on. This is almost never what you want, but this behaviour is available in `testify/require`.